src/plugins/external_tool_support/src/ExternalToolManager.cpp

```diff
@@ -343,7 +348,10 @@ void ExternalToolManagerImpl::sl_customToolsLoaded(Task *task) {
     StrStrMap toolPaths;
     foreach (ExternalTool *tool, toolsList) {
         SAFE_POINT(tool, "Tool is NULL", );
-        QString toolPath = addTool(tool);
+        addTool(tool);
+    }
+    foreach (ExternalTool *tool, toolsList) {
+        QString toolPath = addToolToListsAndReturnToolPath(*tool);
```

Comment:
You have a pointer. There is no need to dereference it.

Reply:
- `void foo(int *i);` -- `i` may be nullptr, but
- `void foo(int &i);` -- means that `i` not null. This allows me not to write `SAFE_POINT` inside the function `foo`. I prefer this type signature for my fiunction.

`*tool` doesn't make function call because it is built-in indirection operator. It's cheap (I don't know for sure, these are my assumptions)